TopMovies/Service/MovieService.swift

```diff
@@ -34,10 +34,22 @@ class MovieService: StoreSubscriber {
               .completed(MovieCategory(dto: $0),
                          $0.results.map { Movie(dto: $0) })}
             .forEach(mainStore.dispatch)
-          mainStore.dispatch(MovieCategoriesAction
-                              .completed(categories: categoriesList.map { MovieCategory(dto: $0) }))
+          mainStore.dispatch(CompletedMovieCategoriesAction(
+                              categories: categoriesList.map(MovieCategory.init(dto:)),
+                              moviesRelational: categoriesList
+                                .map { $0.results.map { Movie(dto: $0) } }
+                                .flatMap { $0 }
+                                .reduce(into: MoviesRelational()) { dict, movie in
+                                  dict[movie.id] = movie
+                                },
+                              relational: categoriesList
+                                .reduce(into: [:]) { dict, categoryDTO in
+                                  dict[MovieCategory.ID(value: categoryDTO.name)]
+                                    = categoryDTO.results
+                                    .map { Movie.ID(value: String($0.id)) }
```

Comment:
Let's also rewrite it in a bit more functional way:
```swift
categoryDTO.results
  .map(String.init)
  .map(Movie.ID.init(value:))
```

Reply:
I can't write .map(String.init), because it will mean String.init(MovieDTO), but i need only id value of MovieDTO
I can do this way:
.map { String($0.id) }
.map(Movie.ID.init(value:))

Reply:
Yeah, I missed that. 
```swift
categoryDTO.results
  .map(\.id)
  .map(String.init)
  .map(Movie.ID.init(value:))
```